### PR TITLE
enable sign-compare warning

### PR DESCRIPTION
**Pull request by dagar**
Enabling `-Wsign-compare` because this caught me the other day.

http://cpp.sh/4pfju

### CODE REVIEW

File: src/modules/logger/logger.cpp [hunk 1]
```diff
@@ -1186,13 +1187,13 @@ int Logger::create_log_dir(tm *tt)
 
 bool Logger::file_exist(const char *filename)
 {
-	struct stat buffer;
+	struct stat buffer = {};
```
Comment (LorenzMeier) [inline src/modules/logger/logger.cpp]: An analysis by Pavel showed that we're spending a major part of the time in PX4 on initialisers. We should not default to them unless we know they are needed.

**Comment by LorenzMeier** [comment]
Nice, thanks! I have a question on the initialisers and obviously this needs testing since you can review it but side-effects are non-trivial.

File: src/modules/logger/logger.cpp (hunk 1 above)
Comment (dagar) [inline src/modules/logger/logger.cpp]: Interesting, I didn't see that. Let's figure out how to approach this more efficiently.

**Comment by dagar** [comment]
I removed zeroing the large buffer for now.

Rolled into https://github.com/PX4/Firmware/pull/7346 to combine testing effort.

**Pull request closed without merging**